pkg/services/ngalert/api/api_provisioning_test.go

```diff
@@ -36,7 +36,20 @@ func TestProvisioningApi(t *testing.T) {
 		require.Equal(t, 202, response.Status())
 	})
 
-	// TODO: we have not lifted out validation yet. Test that we are returning errors properly once validation has been lifted.
+	t.Run("when new policy tree is invalid", func(t *testing.T) {
+		t.Run("POST policies returns 400", func(t *testing.T) {
+			sut := createProvisioningSrvSut()
+			sut.policies = &fakeRejectingNotificationPolicyService{}
+			rc := createTestRequestCtx()
+			tree := apimodels.Route{}
+
+			response := sut.RoutePostPolicyTree(&rc, tree)
+
+			require.Equal(t, 400, response.Status())
+			expBody := `{"error":"invalid policy tree","message":"invalid policy tree"}`
```

Comment:
That particular message comes from a fake, not a real implementation, so it's more generic than what is typical. This test is asserting that the response body contains the exact message that was attached to the validation error, no matter what it is.

In practice, it'll be the specific message of what failed, e.g. `duplicated label <the label> in group_by`, which is a much better message and I think satisfies this request.

Using the same example, I didn't go so far as to identify _which_ policy had the duplicated label, because the existing validation does not do that either, and this PR remained strict as to not changing any currently shipping error messages.